memphis/memphis.py

```diff
@@ -633,7 +633,7 @@ def validate_graphql(self, message):
     def get_dls_msg_id(self, station_name: str, producer_name: str, unix_time: str):
         return station_name + '~' + producer_name + '~0~' + unix_time
 
-    async def produce(self, message, ack_wait_sec: int = 15, headers: Dict[str, Any]={}, async_produce: bool=False, msg_id: Union[str, None]= None):
+    async def produce(self, message, ack_wait_sec: int = 15, headers: Union[Headers, None] = None, async_produce: bool=False, msg_id: Union[str, None]= None):
```

Comment:
Here is example to reproduce this bug.
```python
from random import randint
def f1(dic = {}) -> None:
    dic.update({str(randint(1, 10000)): "foo"})
    print(dic)

f1()
f1()
f1()
```

More reading for exampel here [https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python](url)

Reply:
No, it's should not. Because i add condition to check if this is None.
But because of this header bug, there was chances where even in user did not use headers in produce function, but previous message had headers then we would put wrong previous headers in product without headers passed as arg.

Patching this is not hard just `headers = {}` to `headers = None` and `if headers != {}:` to `if headers is not None:` So this could be done.

```python

# Before change
if headers != {}:

# After change
if headers is not None:
```